### DISCUSSION TITLE
How to reduce the creation time of projects/environments in short-lived containers

### DISCUSSION THREAD

**Discussion by rdubois-mel** [apache/hop#4661] [Q&A] [answered]
Hello,

I use short-lived containers to run my pipelines and workflows, it works like a charm.
But is there a simple way to speed up project and environment creations at the start of execution?

Currently it takes about 30 seconds, not a big deal, but I was wondering if I can optimize this with some configuration.

I tried some jvm settings but without significant improvement.
`--env HOP_OPTIONS="-XX:+AggressiveHeap"`

Thanks,
Remi

**Comment by bamaer** [comment, marked as answer]
the only decent approach is to create your own custom container that only contains the functionality you need, or does not contain what you definitely do not need (I'm looking at you, Beam and Cassandra!). That should already more or less cut your startup time in half. 
Unfortunately, removing functionality from the default image won't shrink your image size and (probably) won't help much in the startup time.

**Comment by rdubois-mel** [comment]
OK thanks,
Indeed, I think I will stay on the official image. But it's good to know for future tests :)
